android/src/com/frostwire/android/gui/activities/ToggleAllSearchEnginesPreference2.java

```diff
@@ -152,32 +144,15 @@ private boolean areAllEnginesChecked() {
         return true;
     }
 
-    private boolean isOneEngineNotChecked() {
-        int count = 0;
-        if (searchEnginePreferences != null) {
-            for (CheckBoxPreference preference : searchEnginePreferences.keySet()) {
-                if (preference != null) {
-                    if (!preference.isChecked()) {
-                        count++;
-                    }
-                }
-            }
-            if (count == 1) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    private boolean isAtLeastOneOtherEngineChecked(CheckBoxPreference checkBoxPreference) {
+    private boolean isTheOnlyCheckedEngine(CheckBoxPreference checkBoxPreference) {
         for (CheckBoxPreference preference : searchEnginePreferences.keySet()) {
             if (preference != null && preference != checkBoxPreference) {
                 if (preference.isChecked()) {
-                    return true;
+                    return false;
                 }
             }
         }
-        return false;
+        return true;
```

Comment:
I see the name suggests that you're verifying that this checkbox preference is the only one that has been checked.

What I see here, is that you return false if some other is checked, but at the end of the method, you don't check if the given checkbox has been checked, perhaps it's not and this is returning true.